crypto/context.c

```diff
@@ -192,21 +183,15 @@ void *openssl_ctx_get_data(OPENSSL_CTX *ctx, int index,
     if (ctx == NULL)
         return NULL;
 
-    CRYPTO_THREAD_read_lock(ctx->lock);
-
     if (ctx->dyn_indexes[index] == -1
-            && !openssl_ctx_init_index(ctx, index, meth)) {
-        CRYPTO_THREAD_unlock(ctx->lock);
+            && !openssl_ctx_init_index(ctx, index, meth))
```

Comment:
note the function openssl_ctx_init_index does something that is forbidden by C standard,
(but allowed by posix): convert a function pointer to void*, it is only guaranteed to be
portable to cast to void (*)()

Reply:
But it is perfectly fine for obfuscated C :smile: